indexer/map_object.cpp

```diff
@@ -96,7 +96,6 @@ std::string MapObject::GetLocalizedType() const
 
 std::string MapObject::GetLocalizedAllTypes(bool withMainType) const
 {
-  ASSERT(!m_types.Empty(), ());
```

Comment:
How/why did it work without your previously merged PR?

Reply:
Something is wrong. Previously, the bool parameter for `FormatSubtitle` was `withType`, it was used to indicate if a type should be queried/formatted by the underlying method. That is why it worked properly (a similar copy-pasted assert in `GetLocalizedType()` wasn't triggered, because `GetLocalizedType` was called only if `withType` was true).

The recent change changed the meaning of `withType` to a different thing: `withMainType` or without a main type, e.g. only secondary types. The case of "no type at all - do not query it" was lost.

Reply:
@biodranik is right, https://github.com/organicmaps/organicmaps/pull/8481 changed the parameter of `FormatSubtitle` and is trying to access the types always. I will try to change this and keep assert